### PR TITLE
[ticket/16524] Fix using out-of-bounds UTF8 characters in profile fields

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

<a href="https://tracker.phpbb.com/browse/PHPBB3-16524">PHPBB3-16524</a>.

### CODE REVIEW

**Comment by 3D-I** [comment]
I propose we use a new permission for this, or **none at all**.

It is better to distinguish the two things, also to avoid criticism as we have already had in the past.

**Comment by rxu** [comment]
Currently used permission has a generic id `u_emoji` so creating another emoji-specific one would be redundant. This way it would end up with own emoji permission for every single text field.

**Comment by 3D-I** [comment]
> This way it would end up with own emoji permission for every single text field.

Exactly, currently only the topic title has permissions for Emoji.
Not any other text field all over the phpBB. We have already covered quite the whole lot.
That's why I think it's best to remove the permission point altogether for this as well

File: phpBB/phpbb/profilefields/type/type_string.php [hunk 1]
```diff
@@ -109,7 +109,7 @@ public function validate_profile_field(&$field_value, $field_data)
 	{
 		/**
 		 * Check for out-of-bounds characters that are currently
-		 * not supported by utf8_bin in MySQL if Emoji is not allowed
+		 * not supported by utf8_bin in MySQL if Emoji are not allowed
```
[the remark this answers is not in the document]
Comment (marc1706) [reply phpBB/phpbb/profilefields/type/type_string.php:112]: Both emoji and emojis is correct, it should however not be written with capital letter

**Comment by marc1706** [comment]
Rebased onto prep-release-3.2.10 to also have it in that ;)

**Comment by 3D-I** [comment]
No milestone?

**Pull request merged**